src/rdkafka_subscription.c

```diff
@@ -92,25 +92,75 @@ rd_kafka_subscribe (rd_kafka_t *rk,
 }
 
 
-rd_kafka_resp_err_t
-rd_kafka_assign (rd_kafka_t *rk,
-                 const rd_kafka_topic_partition_list_t *partitions) {
+rd_kafka_error_t *
+rd_kafka_assign0 (rd_kafka_t *rk,
+                  rd_kafka_assign_method_t assign_method,
+                  const rd_kafka_topic_partition_list_t *partitions) {
         rd_kafka_op_t *rko;
         rd_kafka_cgrp_t *rkcg;
 
         if (!(rkcg = rd_kafka_cgrp_get(rk)))
-                return RD_KAFKA_RESP_ERR__UNKNOWN_GROUP;
+                return rd_kafka_error_new(RD_KAFKA_RESP_ERR__UNKNOWN_GROUP,
+                        rd_kafka_err2str(RD_KAFKA_RESP_ERR__UNKNOWN_GROUP));
```

Comment:
error_new() takes a format string.

And the error string should preferably be something more interesting than err2str, in this case:
"Requires a consumer with group.id configured"